### PR TITLE
6pm c crud tutor

### PR DESCRIPTION
**Pull request by gmundewadi**
This PR creates the tutor table, which has fields for "tutor id", fname, lname, email. It also contains the CRUD operations for a tutor table. There are different tiers of access to this table based on the user’s role. They are as follows: 

- Instructors have edit/delete access to all entries for their course, and view access to all entries for all courses.
- Admins have view/edit/delete access to all entries
- Members have view only access to all entries
- Guests have no access to entries in this table.

This PR also contains backend/frontend tests for all additional code written

### CODE REVIEW

**Review by kourosh-safari — APPROVED** [review]

**Comment by pconrad** [comment]
Please add at least one code review from a member of the team before the staff reviews this.

**Review by tenzingdsherpa — APPROVED** [review]
everything looks good!

**Review by parthatre — COMMENTED** [review]
This PR looks good!

**Review by parthatre — APPROVED** [review]

**Review by tenzingdsherpa — COMMENTED** [review]
looks good!

File: javascript/src/test/components/Tutor/TutorTable.test.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import React, { useCallback } from "react";
+import { render, waitFor } from "@testing-library/react";
+import TutorTable from "main/components/Tutor/TutorTable";
+import userEvent from "@testing-library/user-event";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+// import { useHistory } from "react-router-dom";
```
Comment (kourosh-safari) [inline javascript/src/test/components/Tutor/TutorTable.test.js:7]: Potentially remove these comments from production code.

File: javascript/src/test/pages/Tutor/Tutor.test.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    });
+
+    const { getAllByTestId } = render(<Tutor />);
+    const newButtons = getAllByTestId("new-tutor-button");
+    userEvent.click(newButtons[0]);
+
+    await waitFor(() => expect(pushSpy).toHaveBeenCalledTimes(1));
+  });
+
+  // test("can click to add a tutor", async () => {
```
Comment (kourosh-safari) [inline javascript/src/test/pages/Tutor/Tutor.test.js:210]: potentially remove these comments from prod code

**Review by kourosh-safari — APPROVED** [review]

**Review by tenzingdsherpa — APPROVED** [review]
looks good!

**Review by sethvanb — DISMISSED** [review]
LGTM

File: secrets-localhost.properties.SAMPLE [hunk 3]
```diff
@@ -1,5 +1,5 @@
 app.namespace=https://NAME-OF-YOUR-APP.herokuapp.com
-app.admin.emails=phtcon@ucsb.edu
+app.admin.emails=phtcon@ucsb.edu, longhoang@umail.ucsb.edu,longhoang@ucsb.edu
```
Comment (pconrad) [inline secrets-localhost.properties.SAMPLE:2]: Probably this should not go into the PR.

**Review by pconrad — CHANGES_REQUESTED** [review]
There is one small change that should be made... I can live with it not being made, but it would be good to make it.

**Review by sethvanb — DISMISSED** [review]

File: javascript/src/test/components/Tutor/TutorTable.test.js (hunk 1 above)
Comment (pconrad) [reply javascript/src/test/components/Tutor/TutorTable.test.js:7]: Agreed

File: javascript/src/test/pages/Tutor/Tutor.test.js (hunk 2 above)
Comment (pconrad) [reply javascript/src/test/pages/Tutor/Tutor.test.js:210]: Yes

File: secrets-heroku.properties.SAMPLE [hunk 4]
```diff
@@ -1,9 +1,10 @@
-app.namespace=https://NAME-OF-YOUR-APP.herokuapp.com
+app.namespace=https://ucsb-cs-las-f20-6pm-c.herokuapp.com
 app.admin.emails=phtcon@ucsb.edu
 app.member.hosted-domain=ucsb.edu
 
-auth0.domain=FILL-IT-IN
-auth0.clientId=FILL-IT-IN
+auth0.domain=6pm-c.us.auth0.com
+auth0.clientId=p4W0HAdIJv8tMcRQ6aqLhw4NaSXd26Xg
```
Comment (pconrad) [inline secrets-heroku.properties.SAMPLE:6]: Nooooo!!!!! we don't put secrets in the SAMPLE file that goes into GitHub!  That's the whole point of having a .SAMPLE file that is separate from the real one.

Now these secrets have to be invalidated and we have to generate new ones, doh!

**Review by pconrad — CHANGES_REQUESTED** [review]
I would have been ok with this.. there were minor nitpicks, but I was willing to let them slide.

But checking a secret into the main branch.. that's a bridge too far.

This has to be fixed before it's mergeable, I'm afraid.

**Review by pconrad — APPROVED** [review]
LGTM!

**Comment by pconrad** [comment]
40 pts!

**Pull request merged**